mysql/mysql-lib.pl

```diff
@@ -2181,13 +2181,16 @@ sub mysql_login_type
 }
 
 # list_authentication_plugins()
-# Returns a list of supported authentication plugins for setting passwords
+# Returns a list ref of supported authentication plugins for setting passwords
 sub list_authentication_plugins
 {
 my ($ver, $variant) = &get_remote_mysql_variant();
 if ($variant eq "mariadb" && &compare_version_numbers($ver, "10.4") >= 0 ||
     $variant eq "mysql" && &compare_version_numbers($ver, "5.7.6") >= 0) {
-	return ('mysql_native_password', 'caching_sha2_password', 'unix_socket');
+	my $rv = &execute_sql($master_db, "show plugins");
+	my @plugins = map { $_->[0] } grep { $_->[1] eq 'ACTIVE' &&
+		$_->[2] eq 'AUTHENTICATION' } @{ $rv->{data} };
+	return @plugins ? \@plugins : ['mysql_native_password'];
```

Comment:
Why not return `'caching_sha2_password', 'unix_socket'` as well?

Reply:
Because they are _never_ available in all cases!

Reply:
Ok makes sense!